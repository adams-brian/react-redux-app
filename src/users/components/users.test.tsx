import { shallow } from 'enzyme';
import * as React from 'react';

import EditUser from './editUser';
import UserList from './userList';
import Users from './users';

describe("Users", () => {

  let element: JSX.Element;

  beforeEach(() => {
    element = <Users />
  });

  it('renders as expected', () => {
    const component = shallow(element);
    expect(component).toMatchSnapshot();
  });
  
  it('routes /users to UserList', () => {
    const component = shallow(element);
    expect(component.find('Route[exact=true][path="/users"]').first().prop('component')).toBe(UserList);
  });

  it('routes /users/createuser to EditUser', () => {
    const component = shallow(element);
    expect(component.find('Route[exact=true][path="/users/createuser"]').first().prop('component')).toBe(EditUser);
  });

  it('routes /users/:id to EditUser', () => {
    const component = shallow(element);
    expect(component.find('Route[path="/users/:id"]').first().prop('component')).toBe(EditUser);
  });

});
