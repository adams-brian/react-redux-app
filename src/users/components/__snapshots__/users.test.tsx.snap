// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Users component renders as expected 1`] = `
<Switch>
  <Route
    component={[Function]}
    exact={true}
    path="/users"
  />
  <Route
    component={[Function]}
    exact={true}
    path="/users/createuser"
  />
  <Route
    component={[Function]}
    path="/users/:id"
  />
</Switch>
`;

exports[`Users connected component should render as expected 1`] = `
<loading(Users)
  error=""
  load={[Function]}
  loaded={false}
/>
`;

exports[`Users connected component should render as expected 2`] = `
<div
  className="users-container"
>
  <h1
    className="users-header"
  >
    Users
  </h1>
  <table
    className="table table-hover"
  >
    <thead>
      <tr>
        <th>
          First Name
        </th>
        <th>
          Last Name
        </th>
        <th
          style={
            Object {
              "width": "1px",
            }
          }
        />
      </tr>
    </thead>
    <tbody>
      <UserListRow
        deleteUser={[Function]}
        editUser={[Function]}
        key="123"
        user={
          Object {
            "_id": "123",
            "firstname": "abc",
            "lastname": "def",
          }
        }
      >
        <tr
          onClick={[Function]}
        >
          <td>
            abc
          </td>
          <td>
            def
          </td>
          <td>
            <button
              onClick={[Function]}
            >
              <span
                className="fa fa-times"
              />
            </button>
          </td>
        </tr>
      </UserListRow>
    </tbody>
  </table>
  <Link
    className="create-user"
    replace={false}
    to="/users/createuser"
  >
    <a
      className="create-user"
      href="/users/createuser"
      onClick={[Function]}
    >
      <button
        className="btn btn-primary"
        type="button"
      >
        Create User
      </button>
    </a>
  </Link>
</div>
`;
